constructs/service-event-function.ts

```diff
@@ -0,0 +1,129 @@
+import * as cdk from 'aws-cdk-lib';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as lambdaNodeJs from 'aws-cdk-lib/aws-lambda-nodejs';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import * as sqs from 'aws-cdk-lib/aws-sqs';
+import * as events from 'aws-cdk-lib/aws-events';
+import * as eventTargets from 'aws-cdk-lib/aws-events-targets';
+import * as lambdaEventSources from 'aws-cdk-lib/aws-lambda-event-sources';
+import { Construct } from 'constructs';
+import type { FullHandlerDefinition } from '../extract/extract-handlers';
+import { EventHandlerDefinition } from '../handlers';
+
+export interface ServiceEventFunctionProps {
+	role: iam.IRole;
+	definition: FullHandlerDefinition<EventHandlerDefinition>;
+	bundlingOptions?: lambdaNodeJs.BundlingOptions;
+	layers?: lambda.ILayerVersion[];
+	eventBuses: events.IEventBus[];
+}
+
+export class ServiceEventFunction extends Construct {
+	readonly fn: lambdaNodeJs.NodejsFunction;
+	readonly queue: sqs.Queue;
+	readonly dlq: sqs.Queue;
+	readonly definition: FullHandlerDefinition<EventHandlerDefinition>;
+
+	constructor(
+		scope: Construct,
+		id: string,
+		{
+			role,
+			definition,
+			bundlingOptions,
+			layers,
+			eventBuses,
+		}: ServiceEventFunctionProps,
+	) {
+		super(scope, id);
+
+		const timeout = definition.timeout
+			? cdk.Duration.seconds(definition.timeout)
+			: cdk.Duration.seconds(60);
+
+		const maximumAttempts = definition.maximumAttempts ?? 10;
+
+		this.definition = definition;
+
+		this.dlq = new sqs.Queue(this, 'Dlq', {
+			retentionPeriod: cdk.Duration.days(14),
+			receiveMessageWaitTime: cdk.Duration.seconds(20),
+			visibilityTimeout: timeout,
+		});
+
+		this.dlq.grantSendMessages(role);
+		this.dlq.grantConsumeMessages(role);
+
+		this.queue = new sqs.Queue(this, 'Queue', {
+			retentionPeriod: cdk.Duration.days(14),
+			receiveMessageWaitTime: cdk.Duration.seconds(20),
+			visibilityTimeout: timeout,
+			deadLetterQueue: {
+				maxReceiveCount: maximumAttempts,
+				queue: this.dlq,
+			},
+		});
+
+		let eventBus = eventBuses[0];
+		if (definition.eventBusName) {
+			const matchedEventBus = eventBuses.find(
+				(bus) => bus.eventBusName === definition.eventBusName,
+			);
+			if (!matchedEventBus) {
+				throw new Error(`
+					Could not find the event bus "${definition.eventBusName}" specified event bus name. 
+					Please make sure the event handler "${definition.name}" is configured properly or that you have configured the appropriate event buses.
+				`);
+			}
+			eventBus = matchedEventBus;
+		}
```

Comment:
IMO this logic should be happening inside of `lambda-service.ts` instead of here, and this construct should just take in an `IEventBus` property for the event bus it should be connected to.

Reply:
I see your point, I'll make the change.

Reply:
Addressed in https://github.com/faceteer/cdk/pull/16/commits/feb795380c1ed9c2aec117471829b8a3518e57fc